javascript/ql/test/query-tests/Security/CWE-601/ClientSideUrlRedirect/react.js

```diff
@@ -15,3 +15,23 @@ class Application extends React.Component {
 };
 
 export default Application
+
+import Link from 'next/link'
+export function NextLink() {
+    return <Link href={document.location.hash}><a>this page!</a></Link>;
+}
+
+import { useRouter } from 'next/router'
+
+export function nextRouter() {
+  const router = useRouter();
+  return <span onClick={() => router.push(document.location.hash)}>Click to XSS 1</span>
```

Comment:
Slightly unfortunate example since `.hash` starts with `#` and so can't use the `javascript:` scheme. Same below.
```suggestion
  return <span onClick={() => router.push(document.location.hash.substr(1))}>Click to XSS 1</span>
```